core_backend/tests/api/test_question_answer.py

```diff
@@ -134,6 +139,96 @@ def test_llm_response(
             assert len(content_response) != 0
 
 
+class TestErrorResponses:
+    VALID_LANGUAGE = IdentifiedLanguage.get_supported_languages()[-1]
+
+    @pytest.fixture
+    def user_query_response(
+        self,
+    ) -> UserQueryResponse:
+        return UserQueryResponse(
+            query_id=124,
+            content_response={},
+            llm_response=None,
+            feedback_secret_key="abc123",
+            debug_info={},
+            state=ResultState.IN_PROGRESS,
+        )
+
+    @pytest.fixture
+    def user_query_refined(self, request: pytest.FixtureRequest) -> UserQueryRefined:
+        if hasattr(request, "param"):
+            language = request.param
+        else:
+            language = None
+        return UserQueryRefined(
+            query_text="This is a basic query",
+            original_language=language,
+            query_text_original="This is a query original",
+        )
+
+    @pytest.mark.parametrize(
+        "user_query_refined,should_error",
+        [("ENGLISH", False), ("MADE_UP_LANGUAGE", True), (VALID_LANGUAGE, False)],
+        indirect=["user_query_refined"],
+    )
+    async def test_translate_error(
+        self,
+        user_query_refined: UserQueryRefined,
+        user_query_response: UserQueryResponse,
+        should_error: bool,
+        monkeypatch: pytest.MonkeyPatch,
+    ) -> None:
+        async def mock_ask_llm(*args: Any, **kwargs: Any) -> str:
+            return "This is a translated LLM response"
+
+        monkeypatch.setattr(
+            "core_backend.app.llm_call.parse_input._ask_llm_async", mock_ask_llm
+        )
+        query, response = await _translate_question(
+            user_query_refined, user_query_response
+        )
+        if should_error:
+            assert isinstance(response, UserQueryResponseError)
+            assert response.error_type == ErrorType.UNKNOWN_LANGUAGE
+        else:
+            assert isinstance(response, UserQueryResponse)
+            if query.original_language == "ENGLISH":
+                assert query.query_text == "This is a basic query"
+            else:
+                assert query.query_text == "This is a translated LLM response"
+
+    @pytest.mark.parametrize(
+        "classification, should_error",
+        [("SAFE", False), ("INAPPROPRIATE_LANGUAGE", True), ("PROMPT_INJECTION", True)],
+    )
+    async def test_unsafe_query_error(
```

Comment:
I have not added a test for `paraphrase` since I will be fixing it in the next PR anyway.
I'll add one then.